pkg/auth/image_proxy_test.go

```diff
@@ -55,14 +63,26 @@ func TestFromImagePull(t *testing.T) {
 	assert.NoError(err)
 
 	// Mocking the end CRI request consumer.
+	serverAlpha := &MockAlphaImageService{}
 	server := &MockImageService{}
-	runtime_alpha.RegisterImageServiceServer(mockRPC, server)
-	go mockRPC.Serve(lm)
-	defer lm.Close()
+	runtime_alpha.RegisterImageServiceServer(mockRPC, serverAlpha)
+	runtime.RegisterImageServiceServer(mockRPC, server)
+	go func() {
+		err := mockRPC.Serve(lm)
+		assert.NoError(err)
+	}()
+	defer func() {
+		mockRPC.Stop()
+	}()
 
 	AddImageProxy(ctx, proxyRPC, mockSocket)
-	go proxyRPC.Serve(lp)
-	defer lp.Close()
+	go func() {
```

Comment:
Maybe `go assert.NoError(proxyRPC.Serve(lp))`, `defer proxyRPC.Stop()` ?

Reply:
I tried, but it seems that `proxyRPC.Serve(lp)` be first evaluated before the new goroutine starts. So the test blocks on `proxyRPC.Serve(lp)`